classes/helper.php

```diff
@@ -367,4 +372,27 @@ public static function regular_expression_search(string $search, string $table,
         }
         return $results;
     }
+
+    /**
+     * Replace text in a table and column.
+     *
+     */
+    public static function replace_string(string $table, string $column, int $id, string $match, string $replace): void {
+        global $DB;
+
+        // Skip if any of the parameters are empty.
+        if (empty($table) || empty($column) || empty($id) || empty($match) || empty($replace)) {
+            debugging("Skip replacing text in $table.$column with id $id. One or more parameters are empty.");
+            return;
+        }
+
+        // Get the text in the database.
+        $record = $DB->get_record($table, ['id' => $id], $column);
+
+        // Replace the text.
+        $text = str_replace($match, $replace, $record->$column);
```

Comment:
pros and cons of using $DB->replace_all_text ? I assume using it will be faster and reduce the DB io quite a lot.

Reply:
ah, we can't use this because we can't pass in a where clause to limit it to 1 id. But I think we should clone this function and add the filter, and make a not when we go back to upstream this into core to combine this back ine